sql/sql_table.cc

```diff
@@ -9048,6 +9048,27 @@ mysql_prepare_alter_table(THD *thd, TABLE *table,
         goto err;
       case Alter_drop::FOREIGN_KEY:
         // Leave the DROP FOREIGN KEY names in the alter_info->drop_list.
+	/* If we are dropping a foreign key and expect it to exists,
```

Comment:
Grammar: to exist

Maybe even better would be to use the SQL syntax: "If this is DROP FOREIGN KEY without IF EXIST,"